tests/nativeapp/test_artifacts.py

```diff
@@ -470,6 +476,76 @@ def test_bundle_map_disallows_absolute_dest(bundle_map):
         bundle_map.add(PathMapping(src="app", dest=str(absolute_dest)))
 
 
+def test_bundle_map_disallows_clobbering_parent_directories(bundle_map):
+    # one level of nesting
+    with pytest.raises(TooManyFilesError):
+        bundle_map.add(PathMapping(src="snowflake.yml", dest="./app/"))
+        # Adding a new rule to populate ./app/ from an existing directory. This would
+        # clobber the output of the previous rule, so it's disallowed
+        bundle_map.add(PathMapping(src="./app", dest="./"))
```

Comment:
Is using `./` in an src peeking into the deploy root here? I'm not fully understanding this test or how things get clobbered, it seems to me that this test simply copies `snowflake.yml -> app/snowflake.yml` and `app/* -> ./*`.

Reply:
This rule says "in the deploy root, first copy snowflake.yml to a directory named `app`, then create a directory named `app` and make it identical to `app` from the project root." So in principle snowflake.yml could be lost. If the rules are flipped it's OK though, because first we create the directory, then we add a file to it.

Reply:
To be 100% clear, if the rule was app/* -> app, then it'd be fine too as long as snowflake.yml doesn't get multiple matches.

Reply:
Ah yes, missed the fact that dest is `./` and not `.`. Makes sense